modules/gallery/src/commonMain/kotlin/org/jetbrains/compose/storytale/gallery/material3/FullStoryView.kt

```diff
@@ -0,0 +1,245 @@
+import androidx.compose.material3.DrawerValue
+import androidx.compose.material3.FilledTonalIconButton
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.material3.adaptive.currentWindowAdaptiveInfo
+import androidx.compose.material3.rememberDrawerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.toRoute
+import androidx.window.core.layout.WindowWidthSizeClass
+import kotlinx.coroutines.launch
+import org.jetbrains.compose.storytale.Story
+import org.jetbrains.compose.storytale.gallery.ui.theme.UseCustomDensity
+import org.jetbrains.compose.storytale.storiesStorage
+
+@Composable
+fun FullStorytaleGallery(
+    appState: StorytaleGalleryAppState,
+    navController: NavHostController = rememberNavController(),
+    initialStory: Story? = storiesStorage.firstOrNull(),
+) {
+    val drawerState = rememberDrawerState(initialValue = DrawerValue.Closed)
+
+    val activeStoryItem = remember {
+        mutableStateOf(initialStory?.let { StoryListItemType.StoryItem(it) })
+    }
+
+    LaunchedEffect(Unit) {
+        navController.currentBackStack.collect { backstack ->
+            val entry = backstack.lastOrNull()
+            val isStory = entry?.destination?.route?.startsWith("story/") == true
+            if (entry != null && isStory) {
+                val story = entry.toRoute<StoryScreen>()
+                activeStoryItem.value = storiesStorage.firstOrNull {
+                    it.name == story.storyName
+                }?.let {
+                    StoryListItemType.StoryItem(it)
+                }
+            }
+        }
+    }
+
+    ResponsiveNavigationDrawer(
+        drawerState = drawerState,
+        drawerContent = {
+            DrawerContent(appState, drawerState, navController, activeStoryItem.value)
+        },
+        content = {
+            NavHost(
+                navController = navController,
+                startDestination = StoryScreen(activeStoryItem.value?.story?.name ?: ""),
+                enterTransition = { fadeIn() },
+                exitTransition = { fadeOut() },
+                popEnterTransition = { fadeIn() },
+                popExitTransition = { fadeOut() },
+            ) {
```

Comment:
It might be necessary to set a background color for the NavHost or set the transition to `XXXXTransition.None`, otherwise, there seems to be a layer of white flashing, which is especially noticeable in night mode

Reply:
Got it. I disabled the transitions for now (XXXXTransition.None)